solutions/ceoi-12-JobScheduling.mdx

```diff
@@ -0,0 +1,106 @@
+---
+id: ceoi-12-JobScheduling
+source: CEOI 2012
+title: Job Scheduling
+author: Chuyang Wang
+---
+
+We can use binary search to easily find the minimum number of machines needed for finishing the jobs within the given deadline. 
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N \log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+#define all(x) begin(x), end(x)
+#define mp make_pair
+
+using pi = pair<int, int>;
+using vi = vector<int>;
+
+// test if it is possible to finish the jobs using given # of machines
+// return: first: possible or not, second: if possible, the schedule for the jobs
+pair<bool, vector<vi>> isFeasible(const vector<pi> &jobs, int N, int D, int machineCount)
```

Comment:
make N and D global so you don't have to pass constant parameters

Reply:
doesn't matter that much, i for one like it this way
also just asking isn't `N` just `jobs.size()`?

Reply:
Done